assets/js/components/ExecutionResults/CheckResultDetail/CheckDetailHeader.jsx

```diff
@@ -28,15 +30,15 @@ function CheckDetailHeader({
           <span className="font-medium">{checkDescription}</span>
         </h1>
       </div>
-      {cloudProvider === UNKNOWN_PROVIDER && (
+      {targetCluster && cloudProvider === UNKNOWN_PROVIDER && (
```

Comment:
This makes me think of using the map pattern mentioned previously, as `cloudProvider` cannot both be `UNKNOWN_PROVIDER` _and_ `VMWARE_PROVIDER` a the same time :thinking:

Reply:
I am not seeing how the fact that `cloudProvider` cannot both be `UNKNOWN_PROVIDER` _and_ `VMWARE_PROVIDER` relates to the map pattern being better :thinking:, I mean the end result is the same. What am I missing?

However, for consistency I am ok in changing it that way :+1:

Reply:
So we currently have two separated statements, however they actually both depend on `cloudProvider`:

```javascript
      {targetCluster && cloudProvider === UNKNOWN_PROVIDER && (
        <WarningBanner>
          The following results are valid for on-premise bare metal platforms.
          <br />
          If you are running your HANA cluster on a different platform, please
          use results with caution
        </WarningBanner>
      )}
      {targetCluster && cloudProvider === VMWARE_PROVIDER && (
        <WarningBanner>
          Configuration checks for HANA scale-up performance optimized clusters
          on VMware are still in experimental phase. Please use results with
          caution.
        </WarningBanner>
      )}
```

Since we see that the only condition is `cloudProvider`, to me the semantics of it are a bit different. We could show this logic a little more semantically explicit:

```javascript
// Somewhere above...
const toWarningBanner = {
  [UNKNOWN_PROVIDER]: (<WarningBanner>
                         The following results are valid for on-premise bare metal platforms.
                         <br />
                         If you are running your HANA cluster on a different platform, please
                         use results with caution
                       </WarningBanner>),
  [VMWARE_PROVIDER]: (<WarningBanner>
                        Configuration checks for HANA scale-up performance optimized clusters
                        on VMware are still in experimental phase. Please use results with
                        caution.
                      </WarningBanner>)
}

// ...inside `return` statement
{targetCluster && toWarningBanner[cloudProvider] ?? null}
```

I guess it's a more of a semantic/pedantic style philosophy. I don't actually have strong feelings on it (it could even be a `switch` statement or `if` for example, or not change at all), that's why I left it as a comment and not a review :stuck_out_tongue_winking_eye:

Reply:
Don't get me wrong, I like the suggestion and I will apply it. Just wanted to further understand 😄

Reply:
Now that I am touching this I remember why I like the suggestion: we already had this sort of exchange in  https://github.com/trento-project/web/pull/1371 where I proposed something similar :smile: see commits https://github.com/trento-project/web/pull/1371/commits/fba877831799a6b8d5bc7e3eaa3e517be852ccff and https://github.com/trento-project/web/pull/1371/commits/2bbb5e7049687e63e53a37d6eafe090b609f24a5

However, it ended up how it ended up and I don't want to re-open that pandora box.
The best I can do is https://github.com/trento-project/web/pull/1809/commits/b6fcc693b72377b9901977d20f12826f30f140d6

Reply:
@nelsonkopliku you don't need the `null` statement

Reply:
ah! thanks!